plugins/http/test/expected.data.yml

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright 2021 SkyAPM org
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+segmentItems:
+  {{- range .segmentItems }}
+  {{- if or (eq .serviceName "http-client") (eq .serviceName "http-server")}}
```

Comment:
I have a feeling, there is abuse about `if or`. Such as we use multiple services' segments together, and mixing component names, operation names, with or without reference.
We should make sure the expected data as more strict as possible. Check how we check in the plugin test tool, all things should be separated and checked one by one.

Reply:
done

Reply:
via `ghcr.io/apache/skywalking-agent-test-tool/mock-collector:b01d4137fed862338d78a73b23435ccf425f9962`